### DISCUSSION TITLE
Is there a stable version of the aws-sdk-php ?

### DISCUSSION THREAD

**Discussion by mhcwebdesign** [aws/aws-sdk-php#2965] [Q&A] [answered]
In our OpenCart project we use the aws-sdk-php at https://github.com/opencart/opencart/tree/master/upload/system/storage/vendor/aws/aws-sdk-php, and one of our maintainers applies upgrades to it from this repository almost on a daily basis, at the moment we are at version 3.316.5. Question: Is there a stable version of the aws-sdk-php which supports PHP 7.4 and PHP 8.x versions, which doesn't change so frequently?

**Comment by yenfryherrerafeliz** [comment, marked as answer]
Hi @mhcwebdesign, thanks for reaching out. We are constantly delivering minor updates with new features or bug fixes, but as of now we have a stable GA major version which is 3.x.x Having latest updates from the PHP SDK package is important because is how you can benefit from new service features and bug fixes, but, you could also pin your version to any version you want. For example, if you are on 3.316.5 and that version supports PHP runtime 7.2.5 and over, which meets your requirements around PHP runtime support, then you could just lock that version in your composer file by changing your versioning rule for that dependency as in the following example:

```json
{
    "name": "test/test-project",
    "homepage": "homepage",
    "description": "A brief description",
    "keywords": [""],
    "type": "library",
    "license": "Apache-2.0",
    "authors": [
        {
            "name": "Test Author",
            "homepage": "homepage"
        }
    ],
    "require": {
        "php": ">=7.2.5",
        "aws/aws-sdk-php": '3.316.5'
    },
    "require-dev": {
        ... your dev dependencies
    }
}
```

However, as I mentioned before, it is important to keep up to date on this package so you do not miss any bug fix or any new service feature we release.

Thanks!

**Comment by mhcwebdesign** [comment]
Thank you for your reply. Our OpenCart 4 project is still subject to development and bugfixes, but once we have tagged a final release we'll just go with whatever available aws-sdk-php 3.x.x there will be by then. It's mainly for 3rd party OpenCart extension which may use your library, and we just want to be reasonably sure that there won't be any major bugs or API changes.

**Comment by mhcwebdesign** [comment]
BTW.: Can you give us a realistic estimate as to when can we expect a final stable release of the aws-sdk-php? It is a common practice in the IT industry to do a public beta-release first (that's how your current daily updates and bugfixes look like to me), and than do the final stable release.

**Comment by yenfryherrerafeliz** [comment]
Hi @mhcwebdesign, as I mentioned in my first comment, we have a GA major version for the AWS SDK for PHP, that you can confirm [here](https://docs.aws.amazon.com/sdkref/latest/guide/version-support-matrix.html),  which is 3.x. However, we are constantly providing our customers with new features that they can use and also bug fixes that MUST be done, and hence we can not stop delivering updates to our package. But, you have the option to lock your dependencies to the version you wish through composer. 

I will close this discussion for now since there is not more action items from our side.

Thanks!

[a comment by an automated account is not shown]
